main.go

```diff
@@ -64,12 +89,45 @@ func resyncAll(config *rest.Config) error {
 func main() {
 	flag.Parse()
 
+	// Create OS signal channels.
+	sigs := make(chan os.Signal, 1)
+	stop := make(chan struct{})
+	signal.Notify(sigs, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-sigs
+		glog.V(2).Info("Stopping informers")
+		close(stop)
+		os.Exit(0)
```

Comment:
I try to avoid `os.Exit(0)` whenever possible because it skips deferred functions. For handling SIGTERM, I prefer to return a 0 exit code by falling off the end of `main()`, which follows the Go idiom of the happy path flowing all the way through the function:

https://github.com/GoogleCloudPlatform/kube-metacontroller/blob/94c57bb47f286164b967b983421131cac2faad2e/examples/jsonnetd/main.go#L88-L93

In this case, that would mean moving the infinite poll loop into a goroutine, and breaking when the `stop` channel is ready. We'd also want a `done` channel that gets closed when the polling goroutine returns.

What do you think? I could also restructure this part after merging if you prefer.

Reply:
That makes sense. To be honest I added the whole signal handling part just to be able to [throw *something* at the stop channel](https://github.com/kubernetes/client-go/blob/ad1ee6b/informers/factory.go#L81). I'm aware doing this without even waiting defies the purpose.

I can either pass a dummy channel to the `sharedInformerFactory` and remove this entire routine, or duplicate what you did in the jsonnetd example. What are you leaning towards to?

Reply:
Ah, in that case let's just pass `nil` to the stop channel for now, to keep this PR solely about the generated client/informer. We can add graceful signal handling in a separate PR. Let me know if you plan to do that, or else I'll whip something up.

Reply:
I'll pass `nil` then. Feel free to deal with the signals, you seem to have a clear idea of how you want to implement that stop -> done -> exit logic.